src/main/resources/GoogleApps/GoogleAppsConfigSheet.xml

```diff
@@ -43,13 +42,92 @@
   <syntaxId>xwiki/2.0</syntaxId>
   <hidden>true</hidden>
   <content>{{velocity}}
-## You can modify this page to customize the presentation of your object.
-## At first you should keep the default presentation and just save the document.
+$xwiki.jsx.use('GoogleApps.GoogleAppsConfigClass')
+$xwiki.ssx.use('GoogleApps.GoogleAppsConfigClass')
 
-#set($class = $doc.getObject('GoogleApps.GoogleAppsConfigClass').xWikiClass)
-#foreach($prop in $class.properties)
-  ; $prop.prettyName
-  : $doc.display($prop.getName())
+#set($configClassName="GoogleApps.GoogleAppsConfig")
+#set($formId = "${section.toLowerCase()}_${configClassName}")
+#set($configDoc = $xwiki.getDocument($configClassName))
+#set($className="GoogleApps.GoogleAppsConfigClass")
+#set($prefix="${configDoc.fullName}_${className}_0")
+#set($obj = $configDoc.getObject($className))
+#macro(t $n)##
+    $services.localization.render("GoogleApps.GoogleAppsConfigClass_${n}")##
+#end
+#macro(tp $n $p)##
+    $services.localization.render("GoogleApps.GoogleAppsConfigClass_${n}",$p)##
 #end
-{{/velocity}}</content>
+
+{{html wiki=true}}
+&lt;form id="$formId" method="post" action="$xwiki.getURL($configDoc, 'saveandcontinue')" onsubmit="cancelCancelEdit()" class="xform"&gt;
+    ; $configDoc.display("activate", 'edit', $obj) &lt;label for="${prefix}_activate"&gt;#t("activate")&lt;/label&gt;
+
+    &lt;fieldset&gt;
+        &lt;legend&gt;#t("communicate")&lt;/legend&gt;
+
+        ; &lt;span class="xHint"&gt;##
+           #tp("communicate.hint",  ['[[', '&gt;&gt;https://accounts.google.com/ManageDomains]]'])&lt;br&gt;
+           #tp("communicate.hint2", ['[[', '&gt;&gt;https://store.xwiki.com/xwiki/bin/view/Extension/GoogleAppsIntegration#installation]]'])&lt;/span&gt;
+
+
+        ; &lt;label for="${prefix}_clientid"&gt;#t("clientid")&lt;/label&gt; ##
```

Comment:
In view mode, the `label for=` represents invalid XHTML syntax, as the target input is not displayed (The value of the `for` attribute of the label element must be the ID of a non-hidden form control.)

Reply:
Hello Alex,
can you be more specific?
The input element of clientid is not invisible. It may be disabled, but I think that this is ok.
I've run the validator and only found a duplicate of the id of the form... which can be easily avoided.

Reply:
For this one I got confused by the behavior of the application. I tested it and the form wasn't editable, only displayed in read-only, without inputs to fill. This was caused maybe by the errors in the xar the the fact that the class page was not installed (I added another comment about that).

Reply:
I think this is solved (the screenshot you sent later was indeed a class import issue).
paul